File: src/main/java/se/oeman/SensorReceiverActor.java
package se.oeman;

import akka.actor.UntypedActor;

public class SensorReceiverActor extends UntypedActor {

  @Override
  public void onReceive(Object message) throws Exception {
    if (message instanceof SensorData.Temperature) {
      SensorData.Temperature t = (SensorData.Temperature) message;
      System.out.println("Received temperature:" + t.getTemp());
    } else if (message instanceof SensorData.Humidity) {
      SensorData.Humidity h = (SensorData.Humidity) message;
      System.out.println("Received humidity:" + h.getHumidity());
    }
  }
}

File: src/main/scala/se/oeman/SensorReceiverActor.scala
package se.oeman

import akka.actor.Actor

class SensorReceiverActor extends Actor {
  override def receive = {
    case Temperature(t, p, m, d) =>
      println(s"$d Sensor: $p $m Temperature: ${t}C")
    case Humidity(h, p, m, d) =>
      println(s"$d Sensor: $p $m Humidity: ${h}%")
  }
}
